lua/astrocommunity/recipes/heirline-tabline-buffer-number/init.lua

```diff
@@ -0,0 +1,16 @@
+return {
+  "heirline.nvim",
+  opts = function(_, opts)
+    -- overwrite https://github.com/AstroNvim/AstroNvim/blob/main/lua/astronvim/plugins/heirline.lua#L114
+    local status = require "astroui.status"
+    local function my_tabline_file_info()
+      local tmp = status.component.tabline_file_info()
+      table.insert(tmp, 2, {
+        provider = function(self) return self and self.bufnr and self.bufnr or "" end,
+        hl = { bold = true, underline = true },
+      })
+      return tmp
+    end
+    opts.tabline[2] = status.heirline.make_buflist(my_tabline_file_info())
+  end,
+}
```

Comment:
```suggestion
return {
  "heirline.nvim",
  specs = {
    {
      "AstroNvim/astroui",
      ---@type AstroUIOpts
      opts = {
        status = {
          attributes = {
            bufnr = { bold = true, underline = true },
          },
        },
      },
    },
  },
  opts = function(_, opts)
    -- overwrite https://github.com/AstroNvim/AstroNvim/blob/main/lua/astronvim/plugins/heirline.lua#L114
    local status = require "astroui.status"

    opts.tabline[2] = status.heirline.make_buflist(
      status.component.tabline_file_info { bufnr = { hl = status.hl.get_attributes "bufnr" } }
    )
  end,
}
```

I just released AstroUI which adds a `bufnr` provider as well as adds it as an option to the `file_info` components. Here is an updated block that simplifies this recipe as well as makes stuff like the style configurable by the user through configuring the `attributes` table.

Reply:
It looks like this with this updated recipe:

![2025-04-04_09:14:06_screenshot](https://github.com/user-attachments/assets/27687b2e-2891-430d-8b01-13427f03d940)